apps/logging/loggers.py

```diff
@@ -51,40 +122,60 @@ def log_global_state_metrics(group_timestamp=None):
-                    "id": app.id,
-                    "name": app.name,
-                    "created": app.created,
-                    "updated": app.updated,
-                    "active": app.active,
-                    "first_active": app.first_active,
-                    "last_active": app.last_active,
-                    "require_demographic_scopes": app.require_demographic_scopes,
-                    "real_bene_cnt": real_cnt,
-                    "synth_bene_cnt": synth_cnt,
-                    "user_id": app.user.id,
-                    "user_username": app.user.username,
-                    "user_date_joined": app.user.date_joined,
-                    "user_last_login": app.user.last_login,
-                    "user_organization": getattr(user_profile, "organization_name", None), }
+        access_token_counts = get_access_token_by_app_counts(application=app)
+
+        grant_counts = get_grant_by_app_counts(application=app)
+
+        log_dict = {
+            "type": "global_state_metrics_per_app",
+            "group_timestamp": group_timestamp,
+            "id": app.id,
+            "name": app.name,
+            "created": format_timestamp(app.created),
+            "updated": format_timestamp(app.updated),
+            "active": app.active,
+            "first_active": format_timestamp(app.first_active),
+            "last_active": format_timestamp(app.last_active),
+            "require_demographic_scopes": app.require_demographic_scopes,
+            "real_bene_cnt": grant_counts.get("real_bene_cnt", None),
+            "synth_bene_cnt": grant_counts.get("synth_bene_cnt", None),
+            "real_bene_token_distinct_cnt": access_token_counts.get(
+                "real_bene_distinct_cnt", None
+            ),
+            "synth_bene_token_distinct_cnt": access_token_counts.get(
+                "synth_bene_distinct_cnt", None
+            ),
+            "real_bene_grant_arch_distinct_cnt": grant_counts.get(
+                "real_bene_arch_distinct_cnt", None
+            ),
+            "synth_bene_grant_arch_distinct_cnt": grant_counts.get(
+                "synth_bene_arch_distinct_cnt", None
+            ),
+            "real_bene_grant_and_arch_union_cnt": grant_counts.get(
+                "real_bene_union_cnt", None
+            ),
+            "synth_bene_grant_and_arch_union_cnt": grant_counts.get(
+                "synth_bene_union_cnt", None
+            ),
+            "user_id": app.user.id,
+            "user_username": app.user.username,
+            "user_date_joined": format_timestamp(app.user.date_joined),
+            "user_last_login": format_timestamp(app.user.last_login),
+            "user_organization": getattr(user_profile, "organization_name", None),
+        }
 
         logger.info(log_dict, cls=DjangoJSONEncoder)
 
         count = count + 1
 
+    elapsed_time = round(datetime.utcnow().timestamp() - start_time, 3)
+
     print("---")
```

Comment:
Do we need these print lines? Will this make parsing this harder?

Reply:
@nbragdon These are just for the purpose of showing the activity that occurred when reviewing the Jenkins console logging.

For example, here is part of the console log for a last run in Sandbox. I replaced some of the numbers with XXXXX.

```

TASK [Display command output: log_global_state_metrics] ************************
ok: [10.246.44.75] => {
    "msg": [
        "---",
        "---RUNNING DJANGO COMMAND:  log_global_state_metrics",
        "---",
        "---",
        "---    Wrote top level log entry:  {'type': 'global_state_metrics', 'group_timestamp': '2021-11-05T07:01:49+00:00', 'real_bene_cnt': 0, 'synth_bene_cnt': 14632, 'global_apps_active_cnt': XXXXX, 'global_apps_inactive_cnt': 0, 'global_apps_require_demographic_scopes_cnt': XXXXX}",
        "---",
        "---",
        "---    Wrote per application log entries:  XXXXX",
        "---",
        "SUCCESS",
        ""
    ]
}

```